api/proto/teleport/userpreferences/v1/userpreferences.proto

```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.userpreferences.v1;
+
+import "google/protobuf/empty.proto";
+import "teleport/userpreferences/v1/assist.proto";
+import "teleport/userpreferences/v1/theme.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/userpreferences/v1;userpreferencesv1";
+
+// UserPreferences is a collection of different user changeable preferences for the frontend.
+message UserPreferences {
+  // assist is the preferences for the Teleport Assist.
+  v1.AssistUserPreferences assist = 1;
+  // theme is the theme of the frontend.
+  Theme theme = 2;
+}
+
+// GetUserPreferencesRequest is a request to get the user preferences.
+message GetUserPreferencesRequest {
+  // username is the username of the owner of the user preferences to get.
+  string username = 1;
+}
+
+// UpsertUserPreferencesRequest is a request to create or update the user preferences.
+message UpsertUserPreferencesRequest {
+  // preferences is the new user preferences to set.
+  UserPreferences preferences = 1;
+  // username is the username of the owner of the user preferences to update.
+  string username = 2;
+}
+
+// UserPreferencesService is a service that stores user settings.
+service UserPreferencesService {
+  // GetUserPreferences returns the user preferences for a given user.
+  rpc GetUserPreferences(GetUserPreferencesRequest) returns (UserPreferences);
```

Comment:
Is it possible to try and avoid returning `UserPreferences` and instead return `GetUserPreferencesResponse` which encapsulates UserPreferences - I tend to find having per RPC Request/Response protos makes it much easier to make non-breaking changes in future.

Reply:
Yeah sure